src/app/src/test/kotlin/org/imperial/mrc/hint/integration/PasswordTests.kt

```diff
@@ -41,6 +36,76 @@ class PasswordTests(@Autowired val restTemplate: TestRestTemplate): IntegrationT
 
     @Test
     fun `request reset password link generates email to disk`() {
+        val lines = requestPasswordResetLinkAndReadEmail()
+
+        assertThat(lines.contains("This is an automated email from HINT. We have received a request to reset the" +
+                " password for the account with\n" +
+                "this email address (test.user@example.com)."))
+
+    }
+
+    @Test
+    fun `can reset password with valid token`() {
+        val lines = requestPasswordResetLinkAndReadEmail()
+        val token = getTokenFromEmailText(lines)
+
+        val map = LinkedMultiValueMap<String, String>()
+        map.add("token", token)
+        map.add("password", "newpassword")
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_FORM_URLENCODED
+
+        var entity = restTemplate.postForEntity<String>("/password/reset-password/",
+                HttpEntity(map, headers))
+        Assertions.assertThat(entity.statusCode).isEqualTo(HttpStatus.OK)
+
+        restTemplate.restTemplate.interceptors.add(AuthInterceptor(restTemplate, "newpassword"))
+        entity = restTemplate.getForEntity<String>("/")
+        Assertions.assertThat(entity.statusCode).isEqualTo(HttpStatus.OK)
```

Comment:
Can we also check that response bodies conform to the schema here? I know until https://vimc.myjetbrains.com/youtrack/issue/mrc-434 is done we can't do this programatically using the JSONValidator but just checking manually as per https://github.com/mrc-ide/hint/pull/28/files#diff-02791b2bdd6c77604f5d206a166f297bR63